### PR TITLE
Add docker compose configuration and docs

### PR DESCRIPTION
**Pull request by djjuhasz**
- Add docker compose configuration for building the Sphinx docs and
  running a local webserver to preview the documentation in a web
  browser
- Add DOCKER.md documentation for using Docker

### CODE REVIEW

File: DOCKER.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Viewing the AtoM docs with Docker Compose
+
+This [Access to Memory](https://www.accesstomemory.org/) documentation can be
+generated and viewed locally in a web browser with the provided Docker Compose
+configuration.  Viewing the documentation locally can be very helpful when
```
Comment (jraddaoui) [inline DOCKER.md]: Nitpick: extra whitespace.

File: DOCKER.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+The docker container uses
+[Sphinx Autobuild](https://pypi.org/project/sphinx-autobuild/) to automatically
+rebuild the Sphinx documents when any files are changed in the atom-docs
+directory.  After saving your changes the web documentation should automatically
+update in your browser.
+
+## When you are done
+
+In the atom-docs directory run `docker-compose stop` to stop the container
```
Comment (jraddaoui) [inline DOCKER.md]: Having only one container, I don't know if `down` would be better than `stop` or if it has other implications.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+WORKDIR /etc/
+RUN mkdir -p /etc/Sphinx/build
+
+RUN apk add --no-cache python3 py3-pip make git
+RUN pip3 install --upgrade pip && \
+    pip3 install git+https://github.com/sphinx-doc/sphinx && \
+    pip3 install sphinx-autobuild && \
+    pip3 install sphinx-rtd-theme
+
+CMD sphinx-autobuild -b html --host 0.0.0.0 --port 80 /etc/Sphinx/source /etc/Sphinx/build
```
Comment (jraddaoui) [inline Dockerfile]: Nitpick: you're already in `/etc/` so you may be able to use shorter relative paths here and in line 3.

File: DOCKER.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+instructions on installing Docker in your operating system.
+
+## Building a viewing the documentation with Docker Compose
+
+1. In the atom-docs directory run `docker-compose up -d` to build and start
+   the atom-docs Docker container
+1. Once the container is built, open a web browser and navigate to
+  `http://localhost:8100`
+
+![Screenshot of AtoM docs in Docker](images/atom_docs_in_docker.png)
```
Comment (jraddaoui) [inline DOCKER.md]: Maybe for another iteration, but it would be great to see the actual theme we use in the docs site.

File: DOCKER.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+generated and viewed locally in a web browser with the provided Docker Compose
+configuration.  Viewing the documentation locally can be very helpful when
+making updates to quickly check for syntax or content errors.
+
+## Prerequistes
+
+**These instructions require a working installation of the Docker Engine**
+
+See [Docker Engine Install](https://docs.docker.com/engine/install/) for
+instructions on installing Docker in your operating system.
```
Comment (jraddaoui) [inline DOCKER.md]: Not sure if we want to link to the Docker Compose docs or if that's included in the Docker Engine Install.

**Review by jraddaoui — APPROVED** [review]
Woohoo! This is great @djjuhasz, thank you!

File: DOCKER.md (hunk 2 above)
Comment (djjuhasz) [reply DOCKER.md]: Huh, I've been using `stop` for my AtoM docker-compose env, but it sounds like `down` removes the containers and networks in addition to stopping the container. 😎  Running `docker-compose up -d` after `down` doesn't seem noticeably slower than after `stop`, so I agree that `down` is probably better.

File: DOCKER.md (hunk 5 above)
Comment (djjuhasz) [reply DOCKER.md]: Added.

File: DOCKER.md (hunk 4 above)
Comment (djjuhasz) [reply DOCKER.md]: Agreed. :)

File: Dockerfile (hunk 3 above)
Comment (djjuhasz) [reply Dockerfile]: Done :)

File: DOCKER.md (hunk 1 above)
Comment (djjuhasz) [reply DOCKER.md]: Fixed

**Review by jraddaoui — APPROVED** [review]

**Pull request merged**